src/components/Switch/Switch.tsx

```diff
@@ -1,184 +1,231 @@
+            selected_disabled: !!value && !!disabled,
+            selected_pressed: !!value && !!pressed,
+            selected_focused: !!value && !!focused,
+            selected: !!value,
+            disabled: !!disabled,
+            hovered: !!hovered,
+            focused: !!focused,
+            pressed: !!pressed,
         },
-    );
+    });
+    const toggleMarginAnimation = useRef(new Animated.Value(value ? 0 : 1)).current;
+    const toggleSizeAnimation = useRef(new Animated.Value(value ? 0 : 1)).current;
+
+    const moveToggle = toggleMarginAnimation.interpolate({
+        inputRange: [0, 1],
+        outputRange: [
+            !value && pressed && !unCheckedIcon
+                ? 0
+                : !value && unCheckedIcon
+                ? size * 0.04
+                : size * 0.2,
+            pressed ? size * 0.74 : size * 0.81,
+        ],
+    });
 
-    const { switchStyle, props } = useMemo(() => {
-        const {
-            checkedColor,
-            onTintColor: _onTintColor,
-            thumbTintColor: _thumbTintColor,
-            ...switchStyles
-        } = componentStyles;
+    const toggleSize = toggleSizeAnimation.interpolate({
+        inputRange: [0, 1],
+        outputRange: [
+            !value && !pressed && !unCheckedIcon
+                ? size * 0.5
+                : value || unCheckedIcon
+                ? size * 0.74
+                : size * 0.84,
+            pressed && !value ? size : pressed && value ? size * 0.81 : size * 0.64,
+        ],
+    });
 
-        const thumbTintColor = value && !disabled ? checkedColor : _thumbTintColor;
-        const onTintColor =
-            value && !disabled ? setColor(checkedColor).alpha(0.5).rgb().string() : _onTintColor;
+    const color = useMemo(
+        () => (value ? trackColor?.true : trackColor?.false),
+        [trackColor?.false, trackColor?.true, value],
+    );
 
-        return {
-            switchStyle: switchStyles,
-            props:
-                version && version.major === 0 && version.minor <= 56
-                    ? {
-                          onTintColor,
-                          thumbTintColor,
-                      }
-                    : Platform.OS === 'web'
-                    ? {
-                          activeTrackColor: onTintColor,
-                          thumbColor: thumbTintColor,
-                          activeThumbColor: checkedColor,
-                      }
-                    : {
-                          thumbColor: thumbTintColor,
-                          trackColor: {
-                              true: onTintColor,
-                              false: onTintColor,
-                          },
-                      },
-        };
-    }, [componentStyles, disabled, value]);
+    useEffect(() => {
+        Animated.timing(toggleMarginAnimation, {
+            toValue: value ? 1 : 0,
+            duration: 300,
+            easing: Easing.bezier(0.175, 0.885, 0.32, 1.275),
```

Comment:
create a const for this outside of the component with the name of MOTION_OVERSHOOT.
Later, we need to put this into tokens